selfdrive/car/volkswagen/values.py

```diff
@@ -71,6 +71,26 @@ class CAR:
   SKODA_SUPERB_MK3 = "SKODA SUPERB 3RD GEN"   # Chassis 3V/NP, Mk3 Skoda Superb and variants
   SKODA_OCTAVIA_MK3 = "SKODA OCTAVIA 3RD GEN" # Chassis NE, Mk3 Skoda Octavia and variants
 
+# Required attributes: mass, wheelbase
+# Optional attributes: steer_actuator_delay, steer_ratio, tire_stiffness
+
+ATTRIBUTES = {
```

Comment:
not sure how I feel about this, since it breaks the pattern we use for all the other cars

Reply:
Another alternative would be to make these classes and use inheritance to share common values, then load the appropriate class by name and call a method on the class (implemented in a base class everything inherits from) which returns a `CarParams` instance.  The class constructor can take params to support as much flexibility as we like (e.g. `eps_modified` for different tuning) but in general I think the value of doing something like this is eliminating future bugs by eliminating branch logic as much as possible.

Reply:
Yes, it's different. However, a [13-going-on-32 item](https://github.com/commaai/openpilot/discussions/20346) if-elif chain (before PQ, before MQBevo, before MEB) seems like an undesirable anti-pattern.

Nothing says other ports couldn't be adapted to use a similar form. In fact, the ATTRIBUTE reading code I proposed could be centralized into `selfdrive/car/interfaces.py` such that it gets used to set up any car-ports that use ATTRIBUTE and leaves alone the ones that haven't been converted.

I am not married to this particular implementation, and am also willing to discuss something like @gregjhogan's proposal. I'm just not a fan of what's there today.

Reply:
Forgot to mention: the if-elif chain style has no protection against falling off the end of get_params() with unset parameters if someone forgets or mistypes a car model name. My implementation requires no touches to interface.py at all to add a new car, and will fail earlier and more clearly if someone forgets to add the right ATTRIBUTEs.